pkg/lib/template/template_loader.go

```diff
@@ -173,6 +177,59 @@ func (loader *Loader) Render(
 			}
 			return data, err
 		},
+		"file_sqlite": func(path, statement string) ([]map[string]interface{}, error) {
+			sqliteFile := filepath.Join(rootDir, path)
+			database, err := sql.Open("sqlite3", sqliteFile)
+			if err != nil {
+				return nil, err
+			}
+			defer database.Close()
+
+			rows, err := database.Query(statement)
+			if err != nil {
+				return nil, err
+			}
+			defer rows.Close()
+
+			columns, err := rows.ColumnTypes()
+			if err != nil {
+				return nil, err
+			}
+
+			data := []map[string]interface{}{}
+
+			for rows.Next() {
+				row := []interface{}{}
+				dataEntry := map[string]interface{}{}
+
+				for _, col := range columns {
+					dataEntry[col.Name()] = new(interface{})
+					row = append(row, dataEntry[col.Name()])
+				}
+
+				err = rows.Scan(row...)
+				if err != nil {
+					return nil, err
+				}
+				if len(row) != len(columns) {
+					return nil, errors.Errorf("len of scanned row (%d) != number of columns (%d)", len(row), len(columns))
+				}
+
+				for idx, d := range row {
+					col := columns[idx]
+
+					rv := reflect.ValueOf(d)
+					switch rv.Elem().Interface().(type) {
+					case nil:
+						dataEntry[col.Name()] = nil
+					}
```

Comment:
What happens without this block? The dataEntry should have TYPE+NIL, which is maybe ok?

Reply:
otherwise the NULL value would be `&nil` which could not be checked in the apitest, before we needed `{{ if eq ( $row.v | marshal ) "null" }}`, now it is easier with `{{ if eq $row.v nil }}`

Reply:
good point!